### PR TITLE
amp-43 404 server logic

### PR DESCRIPTION
**Pull request by wajerrr**
- split `graphql` queries into specific modules / http calls
- added logic for handling of 404 / 500 logic
- changed `canonical` to `article` in our data
- returns dummy error html from our handler
- tests

### CODE REVIEW

File: src/server/graphql/fragments.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        height
+        headline
+        url {
+          canonical
+        }
+      }
+    }
+  }`;
+
+export const fragmentN = `fragment N on Content {
```
Comment (yasgreen93) [inline src/server/graphql/fragments.js]: This is the query for the navigation so feel free to change the naming if you need/want to.

File: src/server/graphql/get-data.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  }
+
+  return dataToReturn;
+};
+
+const getData = (ref) =>
+  processDataFromQueries({
+    canonical: getArticleContent(getArticeQuery(ref)),
+    navigation: getContent(navigationQuery),
+    editorsPick: getContent(editorsPickQuery),
```
Comment (yasgreen93) [inline src/server/graphql/get-data.js]: There will also be a query for the footer when V1 supports menus.

File: src/server/graphql/article-query.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import gql from 'graphql-tag';
+import { fragmentC } from './fragments';
+
+const getArticeQuery = (ref) => gql`
+{
+  canonical :canonical(ref: "${ref}")
```
Comment (dosyara) [inline src/server/graphql/article-query.js]: `canonical :canonical` colon is out of place (it was originally, but let's fix it)

File: src/server/graphql/get-data.js (hunk 2 above)
Comment (wajerrr) [reply src/server/graphql/get-data.js]: lets be aware, and add it when it is ready...

File: src/server/graphql/fragments.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+              canonical
+            }
+          }
+        }
+      }
+    }
+  }`;
+
+export const EditorsPickFragment = `fragment EditorsPick on Content {
+      tegID
```
[the remark this answers is not in the document]
Comment (wajerrr) [reply src/server/graphql/fragments.js]: done

File: src/server/graphql/fragments.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        height
+        headline
+        url {
+          canonical
+        }
+      }
+    }
+  }`;
+
+export const navigationFragment = `fragment Navigation on Content {
```
[the remark this answers is not in the document]
Comment (wajerrr) [reply src/server/graphql/fragments.js]: As we only using specific fragments with queries I have moved specific fragments inside query files

File: src/server/graphql/get-data.js [hunk 6]
```diff
@@ -0,0 +1,66 @@
+import getArticeQuery from './article-query';
+import navigationQuery from './naviagtion-query';
+import editorsPickQuery from './editors-pick-query';
+import client from './graphql-client';
+import HttpError from '../utils/http-error';
+
+// TODO this is a bit shaky but only way I found to get graphQL error code
```
[the remark this answers is not in the document]
Comment (wajerrr) [reply src/server/graphql/get-data.js]: yes, it is strange... it is strange I have googled it for last hour.... and I cant find solution for it.it is just not there...
according to documentation 
https://www.apollographql.com/docs/link/links/http.html#error
it should throw network error for "Server network error: the response has a status of >= 300"
(witch should have statusCode)
but in our case it throws GraphQL error without statusCode :-(

I m not sure if this is a bug or am I missing something...
should I carry on investigating ?

File: src/server/graphql/naviagtion-query.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import gql from 'graphql-tag';
+import { navigationFragment } from './fragments';
+
+const prodNavigation = '/content/b9b22pmtme3mcv43b3neeeh4h5gjh3g8';
+// const stageNavigation = '/content/11i5sf2m5v0hmlknqrthfpvpi0or4peu';
```
[the remark this answers is not in the document]
Comment (wajerrr) [reply src/server/graphql/naviagtion-query.js]: removed

File: src/server/graphql/get-data.js [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+      promise.then((result) => {
+        dataToReturn[key] = result.data || result;
+      })
+    );
+  });
+  await Promise.all(promisesToResolve);
+  return dataToReturn;
+};
+
+export const checkDataForErrors = (data) => {
```
Comment (dosyara) [inline src/server/graphql/get-data.js]: how about renaming it to `throwIfDataErrors` or similar? to indicate what function actually does?

Comment (wajerrr) [reply src/server/graphql/get-data.js]: makes sense... changed

**Pull request merged**